recipes/instinct-cpp/all/conanfile.py

```diff
@@ -0,0 +1,158 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, replace_in_file, rm, rmdir
+from conan.tools.microsoft import check_min_vs, is_msvc, is_msvc_static_runtime
+from conan.tools.scm import Version
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class InstinctCppConan(ConanFile):
+    name = "instinct-cpp"
+    description = ("instinct.cpp provides ready to use alternatives to OpenAI Assistant API"
+                   "and built-in utilities for developing AI Agent applications (RAG, Chatbot, Code interpreter) powered by language models. "
+                   "Call it langchain.cpp if you like.")
+    license = "Apache-2.0"
+    url = "https://github.com/RobinQu/instinct.cpp"
+    homepage = "https://github.com/RobinQu/instinct.cpp"
+    topics = ("llm", "agent", "genai")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_duckdb": [True, False],
+        "with_exprtk": [True, False],
+        "with_pdfium": [True, False],
+        "with_duckx": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_duckdb": True,
+        "with_exprtk": True,
+        "with_pdfium": True,
+        "with_duckx": True,
```

Comment:
It would also be great to have successful compilation logs when turning off some or all of these options, just to check that it works :)